hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneTokenIdentifier.java

```diff
@@ -125,7 +136,13 @@ public OzoneTokenIdentifier fromUniqueSerializedKey(byte[] rawData)
       setStrToSign(WritableUtils.readString(in));
     } else {
       this.tokenType = Type.DELEGATION_TOKEN;
-      setOmCertSerialId(WritableUtils.readString(in));
+      String value = WritableUtils.readString(in);
+      try {
+        UUID.fromString(value);
+        setSecretKeyId(value);
+      } catch (IllegalArgumentException e) {
+        setOmCertSerialId(value);
+      }
```

Comment:
using exception handling for this case is probably going to be expensive.
what about:
```suggestion
      UUID.fromString(value);
      if (this.omCertSerialId == null || this.secretKeyId == null) {
        setSecretKeyId(value);
      } else {
        setOmCertSerialId(value);
      }
```

Reply:
Oh wait this is not right. Why does it call UUID.fromString() and ignore the return value?

Reply:
Because this OzoneTokenIdentifier#toUniqueSerializedKey uses its own raw format instead of OMTokenProto format, which I have no idea how.  In a period of time(7d, max delegation time lifetime) after OM is upgraded to use shared secret key, the same index string would be cerial ID or secreKey ID. So this "  UUID.fromString(value);" used to detect whether it's secretKey, which is a UUID, or a serial ID, which is a number.

Reply:
I was going to suggest to use Integer.isNumeric(). But it throws an exception internally so it ends up the same.